commit 82f34f69a3d5a14d4430ef482fea21ffefa07ec5 (refs/changes/72/3072/6)
Author: Rui Loura <rui@joyent.com>
Date:   2017-12-19T17:07:21+00:00 (1 year, 10 months ago)
    
    AGENT-1086 Want to support *_rackN nic_tag suffixes in config-agent
    AGENT-1090 agent.js modeline conflicts with jsstyle options

diff --git a/CHANGES.md b/CHANGES.md
index 0d444c3..0cd4387 100644
--- a/CHANGES.md
+++ b/CHANGES.md
@@ -1,5 +1,11 @@
 # sdc-config-agent changelog
 
+## 1.6.0
+
+- [AGENT-1086] Support network per rack Manta deployments by allowing 
+  nic_tags with \<tag>_rackNN suffixes to override nic_tags with the same \<tag>
+  (e.g "manta_rack99" overrides "manta").
+
 ## 1.5.0
 
 - [TOOLS-1084] Support sync manifest update through SIGHUP using
diff --git a/Makefile b/Makefile
index 0b7e56c..841fa00 100644
--- a/Makefile
+++ b/Makefile
@@ -19,7 +19,7 @@ JS_FILES	:= $(shell ls *.js) $(shell find cmd lib -name '*.js')
 JSL_CONF_NODE	 = tools/jsl.node.conf
 JSL_FILES_NODE   = $(JS_FILES)
 JSSTYLE_FILES	 = $(JS_FILES)
-JSSTYLE_FLAGS    = -o doxygen
+JSSTYLE_FLAGS    = -t 4 -o doxygen
 SMF_MANIFESTS_IN = smf/manifests/config-agent.xml.in
 
 NODE_PREBUILT_VERSION=v0.10.26
diff --git a/agent.js b/agent.js
index 7f6bc3b..41d9c90 100644
--- a/agent.js
+++ b/agent.js
@@ -152,6 +152,16 @@ async.waterfall([
 				if (nic.nic_tag) {
 					var NIC_TAG = nic.nic_tag.toUpperCase();
 					autoMetadata[NIC_TAG + '_IP'] = nic.ip;
+
+					/*
+					 * If there is a nic tag of the form <name>_RACK<number>,
+					 * it will override any similarly named nic tag (e.g.
+					 * "MANTA_RACK##" will override "MANTA" nic tags).
+					 */
+					if (NIC_TAG.search(/^[A-Z]+_RACK\d+$/) === 0) {
+						NIC_TAG = NIC_TAG.split('_')[0];
+						autoMetadata[NIC_TAG + '_IP'] = nic.ip;
+					}
 				}
 			}
 
